### DISCUSSION TITLE
Suppress auth emulator warning message

### DISCUSSION THREAD

**Discussion by KeisukeNagakawa** [gladly-team/next-firebase-auth#482] [Q&A]
Is it possible to disable warning message below when using firebase emulator and next-firebase-auth?
```
firebase.auth().useEmulator("http://localhost:9099", { disableWarnings: true })
```
ref: 
https://github.com/firebase/firebase-tools/issues/2773

Thanks!

**Comment by kmjennison** [comment]
Good question! In this case, I recommend that you initialize the Firebase JS SDK yourself prior to initializing `next-firebase-auth`.

We don't currently have a built-in way to pass options to `useEmulator` (see source code [here](https://github.com/gladly-team/next-firebase-auth/blob/50f8ced6ebc8cfccd65f75067a1b06fa84d4ce89/src/initFirebaseClientSDK.js#L17)).

**Reply by KeisukeNagakawa** [reply]
@kmjennison Thank you very much. I tried to disable warning message according to your advice but it did not work. I think I am mistakenly taken your advice. I wrote code as below in `initAuth.ts`, is it possible for you to point out the wrong part of it? Thanks!
```typescript
  if (process.env.NEXT_PUBLIC_FIREBASE_AUTH_EMULATOR_HOST) {
    const isServerSide = typeof window === "undefined"
    if (!isServerSide) {
      var app
      if (getApps().length == 0) {
        app = initializeApp({
          apiKey: process.env.NEXT_PUBLIC_FIREBASE_PUBLIC_API_KEY,
          authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
          databaseURL: process.env.NEXT_PUBLIC_FIREBASE_DATABASE_URL,
          projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
        })
      } else {
        app = getApp()
      }
      const host = `https://${process.env.NEXT_PUBLIC_FIREBASE_AUTH_EMULATOR_HOST}`
      const auth = getAuth(app)
      connectAuthEmulator(auth, host, { disableWarnings: true })
    }

    init({
      ...initOption,
      firebaseAuthEmulatorHost:
        process.env.NEXT_PUBLIC_FIREBASE_AUTH_EMULATOR_HOST,
    })
  } else {
    init(initOption)
  }
```

**Reply by kmjennison** [reply]
What's not working for you?

In case it helps, here is some example code of initializing Firebase yourself:
https://github.com/gladly-team/next-firebase-auth/discussions/61#discussioncomment-323977

**Reply by KeisukeNagakawa** [reply]
> What's not
Warning message does not go away. But thanks. I'll read the article you've mentioned.
